### ISSUE TITLE
Use high precision elevation data

### ISSUE THREAD

**Issue by karussell** [graphhopper/graphhopper#384]
Currently we use SRTM or CGIAR both having 90m resolution but a new data set released which has 30m resolution. Still the CGIAR data is not updated and I'm not sure if using 'pure' SRTM is wise. 

BTW: Instead of interpolating via surrounding tiles - maybe we should interpolate while creating an edge with the ele data?

Also it looks it is not yet fully released ... in April 2015 it will release for EU+US and other parts will follow this year.

Data sources:
- http://www2.jpl.nasa.gov/srtm/ -> http://dds.cr.usgs.gov/srtm/version2_1/SRTM30/
- New srtm data [now void filled](https://lpdaac.usgs.gov/nasa_shuttle_radar_topography_mission_srtm_global_1_arc_second_data_released_over_middle_east). See the [SRTM user guide](https://lpdaac.usgs.gov/sites/default/files/public/measures/docs/NASA_SRTM_V3.pdf)
- new and open data set from EU [here](http://www.eea.europa.eu/data-and-maps/data/eu-dem). There seems to be more recent data [here](https://spacedata.copernicus.eu/web/cscda/news/-/asset_publisher/NQThjzBp3dRm/content/copernicus-dem-a-new-digital-elevation-model-for-the-copernicus-programme?_101_INSTANCE_NQThjzBp3dRm_redirect=%2Fweb%2Fcscda%2Fnews).
- new and open data set from [UK](https://environmentagency.blog.gov.uk/2015/09/18/laser-surveys-light-up-open-data/)
- new and open data set from Sweden, license is CC-By (ftp://download-open.lantmateriet.se/GSD-Hojddata%20grid%2050%20plus/nh/Stockholm/Sweref%2099%20TM/geotiff/) but there is an older data set which becomes public in ~2020, see section 2.2. of (ftp://download-open.lantmateriet.se/GSD-Hojddata%20grid%2050%20plus/hdb/Stockholm/Sweref%2099%20TM/ascii-tabell/)
 - http://ned.usgs.gov/faq.html#DATA
 - [OpenDEM](https://opendem.info/)

Interesting tool to merge these different data sets [here](https://gitlab.com/lucadelu/OpenDEM_Generator)

**Comment by NopMap** [comment]
It is not a good idea to use SRTM. The holes and errors in mountainous regions are terrible. Best example is this lake in a valley which has an elevation difference of 400m according to SRTM.
http://www.wanderreitkarte.de/forum/thread.php?board=3&thema=28#2
Perfect for water skiing. :-)

A smaller grid does not improve on missing / nonsensical data.

The CGIAR data has been interpolated for small gaps and filled in with data from other sources. I am not aware that an update to 30m is planned. Even if it was it is a considerable piece of work.

There is also the ASTER data, but according to several analyses it only has a pseudo resolution of 30m, the actual data is comparable to SRTM and contains considerable artifacts.

In short, the best elevation data available for free at the moment is still CGIAR and viewfinderpanoramas.

**Comment by andreaswolf** [comment]
Isn’t the "missing data" problem fixed with the newest release of the data? I think I read that somewhere…

I would need that data in GraphHopper for my diploma thesis, but I did not look into the SRTM integration in general. If anybody could give me a few hints on what needs to be changed, I’d give it a try :-)

**Comment by andreaswolf** [comment]
I started implementing this: https://github.com/andreaswolf/graphhopper/tree/srtm-1deg. Still missing is a fallback to the low-res data for the uncovered areas, but that can be added in the high-res provider class.

**Comment by karussell** [comment]
Somehow missed the discussion here.

@andreaswolf nice, very interesting! As a side question: why is CGIAR or the old res not sufficient in your case? And do you intend to additionally interpolate the SRTM data similar to what people did to get CGIAR?

**Comment by andreaswolf** [comment]
Actually, I did not check if the old resolution would suffice, but I think the more precise the data is, the better the results I get for my thesis will be.

I have to check again, but IIRC with version 2.1 of the SRTM data, all holes that CGIAR removed were also fixed in the original data. Is that what you mean with interpolating the data?

**Comment by karussell** [comment]
> also fixed in the original data. Is that what you mean with interpolating the data?

Yes & interesting. But they just filled holes or also applied some low-pass / high-pass filters or whatever to make the landscape more natural ... maybe @NopMap knows more?

**Comment by karussell** [comment]
@andreaswolf if you like to give me some insights in your thesis (would like to hear what you plan :)) feel free to ping me via email. BTW: really *diploma* thesis, wow & nice ;) ?

**Comment by karussell** [comment]
It looks like the data is [now void filled](https://lpdaac.usgs.gov/nasa_shuttle_radar_topography_mission_srtm_global_1_arc_second_data_released_over_middle_east) opinions? There is also a [SRTM user guide](https://lpdaac.usgs.gov/sites/default/files/public/measures/docs/NASA_SRTM_V3.pdf)

**Comment by karussell** [comment]
And there is a new data set from EU [here](http://www.eea.europa.eu/data-and-maps/data/eu-dem)

**Comment by grischard** [comment]
There's also data from Luxembourg here: https://data.public.lu/en/datasets/bd-l-mnt5/

It would be nice to be able to tell Graphhopper to use more than one local source.

**Comment by boldtrn** [comment]
Maybe the easiest step for us right now, would be to implement another elevation provider [SRTMV3, with preferably 1 arc-second resolution (~30*30m)](https://earthdata.nasa.gov/nasa-shuttle-radar-topography-mission-srtm-version-3-0-global-1-arc-second-data-released-over-asia-and-australia).

You can download the new version [here](https://e4ftl01.cr.usgs.gov/SRTM/). However, the data is a) huge, b) downloads are ridiculously slow, and c) need an account, which makes automatic downloads hard. I am not sure if mirroring the new data is allowed, since they use the ASTER data, redistribution of the original ASTER data is not allowed AFAIK, see the [OSM mailing list](https://lists.openstreetmap.org/pipermail/talk/2009-July/038137.html) and [wiki](https://wiki.openstreetmap.org/wiki/ASTER). Using the data itself to derive work or services should be fine.

**Comment by karussell** [comment]
I think the current code was adapted already to work with a more fine grained grid, but I've also not really tested this on my own in the last months. Also the license seems unclear to me and not open data (?) ASTER is technically nice but license-wise ugly.

> Using the data itself to derive work or services should be fine.

Ugly license that forbids redistribution as we would redistribute it partly (for all roads).

> However, the data is a) huge

Shouldn't be a big problem these days :)

**Comment by boldtrn** [comment]
I have found a very extensive list of possible elevation sources [here](https://github.com/openterrain/openterrain/wiki/Terrain-Data).

**Comment by boldtrn** [comment]
I was looking into the elevation data of [finnland](http://kartat.kapsi.fi/files/korkeusmalli/hila_10m/etrs-tm35fin-n2000/). Understanding the mapping of coordinates to the naming of the files is quite hard, as they don't use coordinates in their name (like SRTM and others). However, I found this pdf explaining [the naming schema](http://maanmittauslaitos.fi/sites/maanmittauslaitos.fi/files/old/UTM_lehtijakopdf.pdf) a bit more.

**Comment by boldtrn** [comment]
[Mapzen's Elevation Data is publicly available on AWS](https://aws.amazon.com/de/public-datasets/terrain/) and will be available even after Mapzen shuts down.

**Comment by msbarry** [comment]
Hi - is there any interest in adding an option for mapzen elevation data provider? It comes in SRTM format with global 1 arc second coverage at `https://s3.amazonaws.com/elevation-tiles-prod/skadi/{N|S}{y}/{N|S}{y}{E|W}{x}.hgt.gz`. I'd be happy to submit a pull request (or open another issue to track) unless there is a reason you'd prefer not to.

**Comment by boldtrn** [comment]
I think adding Mapzen's elevation data would be nice. The data has some holes and glitches from SRTM but the last time I tested it, it looked quite good overall. Not sure if there are any reasons not to add it?

**Comment by karussell** [comment]
If it is the same format we should probably just make it configurable (?)

**Comment by nilsnolde** [comment]
Just to leave it here for future reference, the code generating the tiles is also still there:

https://github.com/tilezen/joerd/blob/master/docs/data-sources.md

**Comment by msbarry** [comment]
configurable would be nice. I prototyped this a while ago. It’s the same format, but IIRC the file naming is slightly different and latitude is not limited (range is -90 to 90)

**Comment by msbarry** [comment]
I took a shot at this in #1929. As it turns out it wasn't exactly the same format as the other SRTM providers (mapzen/skadi data is gzipped, SRTM is zipped, and SRTMLG1 is not compressed at all) and the naming conventions for existing SRTM providers had special cases so it was hard to extract a configuration-based template so #1929 just implements the mapzen/skadi format with a configurable base path.
